java/flight/src/main/java/org/apache/arrow/flight/NoOpFlightProducer.java

```diff
@@ -21,6 +21,9 @@
 
 import org.apache.arrow.flight.impl.Flight.PutResult;
 
+/**
+ * A {@link FlightProducer} that throws on all operations.
```

Comment:
i am not sure why a noop producer throws on all operations :) should we change this (in another jira probably?)

Reply:
@lidavidm?

Reply:
It could be changed later, but the Javadoc is fine. (It's just meant to provide default implementations - guess no-op is misleading for what it does though.)